viam-cartographer.go

```diff
@@ -637,7 +639,18 @@ func (cartoSvc *cartographerService) readDataOnInterval(ctx context.Context, lid
 
 func (cartoSvc *cartographerService) getNextDataPoint(ctx context.Context, lidar lidar.Lidar, c chan int) {
 	if _, err := dim2d.GetAndSaveData(ctx, cartoSvc.dataDirectory, lidar, cartoSvc.logger); err != nil {
-		cartoSvc.logger.Warn(err)
+		if strings.Contains(err.Error(), "reached end of dataset") {
+			if !jobsDone {
+				cartoSvc.logger.Warn(err)
+				_, err = os.Create(filepath.Join(cartoSvc.dataDirectory, "Jobs_Done.txt"))
+				if err != nil {
+					cartoSvc.logger.Warn(err)
+				}
+				jobsDone = true
```

Comment:
1. Could we talk about the requirements for this feature so I can be sure that it gets rolled into full mod?
2. I don't think it works to track a given session completing at the package level, as variable is shared across viam cartographer instances & persists even after reconfiguration. 
3. I recommend instead (assuming this should be scoped to a given viam cartographer instance & should be reset on restart) to put it on thecartographerService struct. 
4. Also, should Jobs_Done.txt still persist if cartographer is reconfigured?
cc @EshaMaharishi

Reply:
It doesn't matter if the jobs done file persists after cartographer is reconfigured since a cloud slam session will only be configured once, but I agree for code cleanliness it's better to scope it to a single configuration

Reply:
followed up offline, in the followup ticket we will switch to using a DoCommand